Source/WebCore/html/FileInputType.cpp

```diff
@@ -289,8 +241,8 @@ void FileInputType::disabledStateChanged()
     auto root = element()->userAgentShadowRoot();
     if (!root)
         return;
-    
-    if (RefPtr button = childrenOfType<UploadButtonElement>(*root).first())
+
+    if (RefPtr button = childrenOfType<HTMLInputElement>(*root).first())
```

Comment:
Can we make this less fragile? Repeating this logic isn't great. There are 2 ways to do this:
- create a helper method (`FileInputType::uploadButtonElement()`?) that returns `childrenOfType<HTMLInputElement>(*root).first()`
- store the `uploadButton` as a member on `FileInputType`

You'll also be able to de-duplicate the logic that gets the `userAgentShadowRoot`and the `element`.